ReactiveCocoa/NSObject+Lifetime.swift

```diff
@@ -15,10 +15,71 @@ extension Reactive where Base: NSObject {
 			let token = Lifetime.Token()
 			let lifetime = Lifetime(token)
 
+			let objcClass: AnyClass = (base as AnyObject).objcClass
+			let deallocSelector = sel_registerName("dealloc")!
+
+			// Swizzle `-dealloc` so that the lifetime token is released at the
+			// beginning of the deallocation chain, and only after the KVO `-dealloc`.
+			objc_sync_enter(objcClass)
+
+			// Swizzle the class only if it has not been swizzled before.
+			if objc_getAssociatedObject(objcClass, &lifetimeKey) == nil {
+				objc_setAssociatedObject(objcClass, &lifetimeKey, true, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+
+				var existingImpl: IMP? = nil
+
+				let newImplBlock: @convention(block) (UnsafeRawPointer) -> Void = { objectRef in
+					// A custom trampoline of `objc_setAssociatedObject` is used, since
+					// the imported version has been inserted with ARC calls that would
+					// mess with the object deallocation chain.
+
+					// Release the lifetime token.
+					_rac_objc_setAssociatedObject(objectRef, &lifetimeTokenKey, nil, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment:
As I said, the block is an implementation that is shared across all instances of the class. Unless you create a subclass for each instance at runtime, this would not work.